services/arrow-flight/src/test/java/com/dremio/service/flight/AbstractTestFlightSqlServer.java

```diff
@@ -52,8 +52,49 @@ public void testGetCatalogs() {
 
       String catalogName = ((VarCharVector) root.getVector("catalog_name")).getObject(0).toString();
       Assert.assertEquals("DREMIO", catalogName);
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testGetSchemasWithNoFilter() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getSchemas(null, null, getCallOptions());
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket(), getCallOptions())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+
+      final String[] expectedSchemas =
+        {"INFORMATION_SCHEMA", "cp", "dacfs", "dfs", "dfs_root", "dfs_test", "sys", "sys.cache"};
+
+      Assert.assertEquals(expectedSchemas.length, root.getRowCount());
+
+      VarCharVector catalogNameVector = (VarCharVector) root.getVector("catalog_name");
+      VarCharVector schemaNameVector = (VarCharVector) root.getVector("schema_name");
+
+      for (int i = 0; i < expectedSchemas.length; i++) {
+        String catalogName = catalogNameVector.getObject(i).toString();
+        String schemaName = schemaNameVector.getObject(i).toString();
+        Assert.assertEquals("DREMIO", catalogName);
+        Assert.assertEquals(expectedSchemas[i], schemaName);
+      }
+    }
+  }
+
+  @Test
+  public void testGetSchemasWithFilters() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getSchemas("DREMIO", "INFORMATION_SCHEMA", getCallOptions());
```

Comment:
Should probably test with catalog filter only, schema filter only as well.

Reply:
Fixed